hedera-node/hedera-app/src/main/java/com/hedera/node/app/store/ReadableStoreFactory.java

```diff
@@ -136,7 +136,7 @@ private static Map<Class<?>, StoreEntry> createFactoryMap() {
      */
     public ReadableStoreFactory(@NonNull final State state) {
         this.state = requireNonNull(state, "The supplied argument 'state' cannot be null!");
-        if (state instanceof MerkleStateRoot merkleStateRoot) {
+        if (state instanceof PlatformMerkleStateRoot merkleStateRoot) {
             this.versionFactory = merkleStateRoot.getVersionFactory();
```

Comment:
Why was `getVersionFactory()` removed from `MerkleStateRoot`?

Reply:
Because this functionality is needed by the platform only. It doesn't have to be a part of the state.